tests/test_invenio_stats.py

```diff
@@ -78,101 +78,74 @@ def test_publish_and_consume_events(app, event_entrypoints):
+    mock_user.get_id = lambda: '123'
+    mock_user.is_authenticated = True
+
+    current_queues.declare()
+
+    with patch('invenio_stats.utils.current_user', mock_user):
+        with app.test_request_context(
+            headers={'USER_AGENT':
+                     'Mozilla/5.0 (Windows NT 6.1; WOW64) '
+                     'AppleWebKit/537.36 (KHTML, like Gecko)'
+                     'Chrome/45.0.2454.101 Safari/537.36'}):
+            ids = [uuid.UUID((
+                '0000000000000000000000000000000' + str(i))[-32:])
+                for i in range(100000)]
+            # ids = [ for i in range(1000 * len(objects))]
+            multiple_events = ids[:1000]
+            multiple_events = [event for event in multiple_events
+                               for _ in range(100)]
+            ids = ids + multiple_events
+            generator_list = []
+            for i in ids:
+                file_obj = objects[0]
+                file_obj.file_id = i
+                file_obj.bucket_id = i
+                msg = dict(
+                    # When:
+                    timestamp=datetime.datetime.utcnow().isoformat(),
+                    # What:
+                    bucket_id=str(i),
+                    file_id=str(i),
+                    filename='test.pdf',
+                    # labels=record.get('communities', []),
+                    # Who:
+                    visitor_id=100,
+                )
+                generator_list.append(msg)
+
+            mock_queue = Mock()
+            mock_queue.consume.return_value = (msg for msg in generator_list)
+            mock_queue.routing_key = 'stats-file-download'
+            mock_processor = MagicMock()
+            mock_processor.run = EventsIndexer(mock_queue,
+                                               'events',
+                                               '%Y-%m-%d',
+                                               current_search_client).run
+            mock_processor.actionsiter = EventsIndexer.actionsiter
+            mock_processor.process_event = EventsIndexer.process_event
+            mock_cfg = MagicMock()
+            mock_cfg.processor = mock_processor
+            mock_events_dict = {'file-download': mock_cfg}
+            mock_events = MagicMock()
+            mock_events.__getitem__.side_effect = \
+                mock_events_dict.__getitem__
+        with patch('invenio_stats.ext._InvenioStatsState.events', mock_events):
+            process_events(['file-download'])
+
+    aggregate_event.delay('file-download')
+
+    current_search_client.indices.flush(index='stats-file-download-2017-07')
+
+    for _id in ids[:1000]:
+        print("checking count of ", _id)
+        query = Search(using=current_search_client,
+                       index='events-stats-file-download').\
+            query('term', file_id=_id)
+        assert query.execute().hits.total == 101
+
+    current_search_client.indices.delete(index='events-stats-file-download')
```

Comment:
You are only deleting the alias here. We can delete all indices with `*` I think